packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambdaLayerWalkthrough.ts

```diff
@@ -144,10 +151,19 @@ export async function updateLayerWalkthrough(
 
     // update layer version based on inputs
     parameters.permissions = layerInputParamsToLayerPermissionArray(layerInputParameters);
-  } else {
-    const defaultPermission: PrivateLayer = { type: PermissionEnum.Private };
-    parameters.permissions = storedLayerParameters.permissions || [defaultPermission];
+
+    // update ephemeral state if deployed version is picked
+    if (selectedVersionNumber) {
+      const { envName }: { envName: string } = context.amplify.getEnvInfo();
+      saveLayerVersionPermissionsToBeUpdatedInCfn(parameters.layerName, envName, selectedVersionNumber, parameters.permissions);
+    }
   }
+
+  if (!permissionsUpdateConfirmed || _.isEqual(permissions, parameters.permissions)) {
+    // Nothing has been updated
+    exitOnNextTick(0);
```

Comment:
We should not exit the process, but change how this function works like retuning undefined or anything that signals the caller to not to do further action just return from the flow.